src/main/java/com/aws/greengrass/clientdevices/auth/api/ClientDevicesAuthServiceApi.java

```diff
@@ -51,21 +51,26 @@ public ClientDevicesAuthServiceApi(SessionManager sessionManager, DeviceAuthClie
      * @param certificatePem PEM encoded client certificate.
      * @return True if the provided client certificate is trusted.
      */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
     public boolean verifyClientDeviceIdentity(String certificatePem) {
-        boolean success;
-
-        // Allow internal clients to verify their identities
-        if (deviceAuthClient.isGreengrassComponent(certificatePem)) {
-            success = true;
-        } else {
-            success = useCases.get(VerifyIotCertificate.class).apply(certificatePem);
+        try {
+            boolean success;
+
+            // Allow internal clients to verify their identities
+            if (deviceAuthClient.isGreengrassComponent(certificatePem)) {
+                success = true;
+            } else {
+                success = useCases.get(VerifyIotCertificate.class).apply(certificatePem);
+            }
+
+            domainEvents.emit(new VerifyClientDeviceIdentityEvent(success ? VerifyClientDeviceIdentityEvent
+                    .VerificationStatus.SUCCESS : VerifyClientDeviceIdentityEvent.VerificationStatus.FAIL));
+
+            return success;
+        } catch (Exception e) {
```

Comment:
We don't want to catch only exception. In fact, runtime exceptions are the most important to catch as a service error because they aren't otherwise handled.

Reply:
Do we want to handle runtime exceptions separately then? I thought that catching exception would also catch runtime exceptions since it inherits from exception

Reply:
I have another question branching off of that, in the IPC handlers a new ServiceError is thrown after they catch an exception, which is why I considered any exception as one. Is that ServiceError different from what we're tracking?

Reply:
Yes, because we're not only tracking IPC invocations, but any invocation including from Moquette.

Reply:
Ah ok, thank you